stats/stats.py

```diff
@@ -0,0 +1,324 @@
+                    'customer_name': customer
+                }
+            ]
+        }
+        stats_list.append(data)
+        return stats_list
+
+    def subscriptions_estimated_income(self, queryset):
+        entries = BillingLog.objects.filter(subscription_id=OuterRef('pk')).order_by().\
+            values('subscription_id').annotate(sum=Sum('total')).values('sum')
+
+        final = queryset.annotate(
+            subscription_amount=Coalesce(ExpressionWrapper(
+                Subquery(entries),
+                output_field=DecimalField(decimal_places=2)
+            ), Decimal('0.00'))
+        ).values(
+            'subscription_amount', 'id', 'plan', 'plan__name', 'customer__first_name',
+            'customer__last_name'
+        ).order_by('-subscription_amount')
+
+        stats_list = []
+        for i in final:
+            if not stats_list:
+                stats_list = self.add_list_item(stats_list, i['plan__name'], i['plan'],
+                                                i['id'], i['subscription_amount'],
+                                                i['customer__first_name'] + " " + i['customer__last_name'])
+            else:
+                ok = 0
+                for item in stats_list:
+                    for key, value in item.iteritems():
+                        if key == 'granulations':
+                            for keys, values in value.iteritems():
+                                if values['name'] == i['plan__name']:
+                                    item['values'].append(
+                                        {
+                                            'subscription_id': i['id'],
+                                            'estimated_income': i['subscription_amount'],
+                                            'customer_name': i['customer__first_name'] + " " +
+                                                                            i['customer__last_name']
+                                        }
+                                    )
+                                    ok = 1
+                if ok == 0:
+                    stats_list = self.add_list_item(stats_list, i['plan__name'], i['plan'], i['id'],
+                                                    i['subscription_amount'],
+                                                    i['customer__first_name'] + " " + i[
+                                                        'customer__last_name'])
+
+        if self.modifier is not None:
+            for i in Plan.objects.exclude(subscription__isnull=False).distinct():
+                data = {'granulations': {
+                    'plan': {
+                        'name': i.name,
+                        'id': i.id
+                    }},
+                    'values': []
+                }
+                stats_list.append(data)
+
+        return stats_list
+
+    def get_result(self):
+        if self.queryset.model == Transaction:
```

Comment:
You should build these dynamically and avoid all the if branches. Have a method that receives a model and returns the model name (e.g. `Transaction` -> `transactions`) and use that to obtain your method name.